display_list/display_list_mask_filter.h

```diff
@@ -0,0 +1,212 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_DISPLAY_LIST_DISPLAY_LIST_MASK_FILTER_H_
+#define FLUTTER_DISPLAY_LIST_DISPLAY_LIST_MASK_FILTER_H_
+
+#include "flutter/display_list/types.h"
+#include "flutter/fml/logging.h"
+
+namespace flutter {
+
+class DlBlurMaskFilter;
+
+// The DisplayList MaskFilter class. This class was designed to be:
```

Comment:
Would it make sense to have an abstract class with this documentation on it, and have these classes implement it?

For example, it might specify a virtual `type`, `size`, `shared`, `skia_shared` or something?